// @aws-sdk/lib-dynamodb/dist-cjs/commands/ExecuteStatementCommand.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExecuteStatementCommand = void 0;
const client_dynamodb_1 = require("@aws-sdk/client-dynamodb");
const smithy_client_1 = require("@aws-sdk/smithy-client");
const utils_1 = require("../commands/utils");
class ExecuteStatementCommand extends smithy_client_1.Command {
    constructor(input) {
        super();
        this.input = input;
        this.inputKeyNodes = [{ key: "Parameters" }];
        this.outputKeyNodes = [{ key: "Items" }];
    }
    resolveMiddleware(clientStack, configuration, options) {
        const { marshallOptions, unmarshallOptions } = configuration.translateConfig || {};
        const command = new client_dynamodb_1.ExecuteStatementCommand(utils_1.marshallInput(this.input, this.inputKeyNodes, marshallOptions));
        const handler = command.resolveMiddleware(clientStack, configuration, options);
        return async () => {
            const data = await handler(command);
            return {
                ...data,
                output: utils_1.unmarshallOutput(data.output, this.outputKeyNodes, unmarshallOptions),
            };
        };
    }
}
exports.ExecuteStatementCommand = ExecuteStatementCommand;
